python/katana/lonestar/analytics/deg_count.py

```diff
@@ -0,0 +1,68 @@
+import numpy as np 
+import pyarrow
+
+from katana.property_graph import PropertyGraph
+from katana.datastructures import LargeArray, AllocationPolicy
+from katana.loops import do_all, do_all_operator
+from katana.atomic import (
+    atomic_add,
+)
+
+@do_all_operator()
+def initialize_in_deg(nin, nid):
+    nin[nid] = 0
+
+@do_all_operator()
+def count_in_and_out_deg(graph: PropertyGraph, nout, nin, nid):
+    outdeg = 0
+    for edge in graph.edges(nid):
+        outdeg += 1
+        dst = graph.get_edge_dst(edge)
+        atomic_add(nin, dst, 1)
+    nout[nid] = outdeg
+
+@do_all_operator()
+def count_weighted_in_and_out_deg(graph: PropertyGraph, nout, nin, weightArr, nid):
```

Comment:
Use `weight_array` or just `weight`. It might also be good to use `in_degree` and `out_degree` for clarity.